src/util/GameLogicUtils.cpp

```diff
@@ -64,6 +64,23 @@ namespace spy::util {
         return isPersonOnField(s, target);
     }
 
+    std::optional<UUID>
+    GameLogicUtils::getPersonOnNeighbourField(const gameplay::State &s, const Point &target, const Point &charCoord) {
```

Comment:
Wieso bekommt man hier nicht den Charakter zurück?
Ergibt doch wenig Sinn die UUID zurückzugeben, wenn du danach in der Verwendung eh, falls der return Wert einen value hat den Charakter suchst?

Reply:
Es ginge maximal ein std::optional<const_iterator>, da es keine optional Referenzen gibt.

Reply:
Ja const_iterator passt doch: Du änderst ja nichts und liest nur Daten aus, wenn ich das richtig sehe.
Würde die Methode dann auch in `findCharacterOnNeighbourField` umbenennen.